pkg/handler/nfts_test.go

```diff
@@ -0,0 +1,236 @@
+	"github.com/gin-gonic/gin"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHandler_GetNewListedNFTCollection(t *testing.T) {
+	cfg := config.LoadTestConfig()
+	db := testhelper.LoadTestDB("../../migrations/test_seed")
+	repo := pg.NewRepo(db)
+	log := logger.NewLogrusLogger()
+	s := pg.NewPostgresStore(&cfg)
+	entityMock := entities.New(cfg, log, repo, s, nil, nil, nil, nil, nil, nil, nil)
+
+	type fields struct {
+		entities *entities.Entity
+		log      logger.Logger
+	}
+	tests := []struct {
+		name             string
+		fields           fields
+		wantCode         int
+		wantErr          error
+		wantResponsePath string
+	}{
+		// TODO: Add test cases.
+		{
+			name: "get succesfully",
+			fields: fields{
+				entities: entityMock,
+				log:      log,
+			},
+			wantCode:         200,
+			wantErr:          nil,
+			wantResponsePath: "testdata/get_nft_recent/200.json",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			h := &Handler{
+				entities: tt.fields.entities,
+				log:      tt.fields.log,
+			}
+			w := httptest.NewRecorder()
+			ctx, _ := gin.CreateTestContext(w)
+			ctx.Request = httptest.NewRequest("GET", "/api/v1/nfts/new-listed", nil)
+
+			h.GetNewListedNFTCollection(ctx)
+			require.Equal(t, tt.wantCode, w.Code)
+			expRespRaw, err := ioutil.ReadFile(tt.wantResponsePath)
+			require.NoError(t, err)
+			require.JSONEq(t, string(expRespRaw), w.Body.String(), "[Handler.GetChains] response mismatched")
+		})
+	}
+}
+
+func TestHandler_GetNFTDetail(t *testing.T) {
+	cfg := config.LoadTestConfig()
+	db := testhelper.LoadTestDB("../../migrations/test_seed")
+	repo := pg.NewRepo(db)
+	log := logger.NewLogrusLogger()
+	s := pg.NewPostgresStore(&cfg)
+
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	indexerMock := mock_indexer.NewMockService(ctrl)
+	entityMock := entities.New(cfg, log, repo, s, nil, nil, nil, nil, indexerMock, nil, nil)
+
+	type fields struct {
+		entities *entities.Entity
+		log      logger.Logger
+	}
+	tests := []struct {
+		name             string
+		fields           fields
+		querySymbol      string
+		queryTokenId     string
+		queryGuildId     string
+		expectedAddress  string
+		wantIndexerResp  *response.IndexerGetNFTTokenDetailResponse
+		wantCode         int
+		wantErr          error
+		wantResponsePath string
+	}{
+		// TODO: Add test cases.
+		{
+			name: "query match single record",
+			fields: fields{
+				entities: entityMock,
+				log:      log,
+			},
+			querySymbol:     "PH",
+			queryTokenId:    "1",
+			queryGuildId:    "",
+			expectedAddress: "0xb54FF1EBc9950fce19Ee9E055A382B1219f862f0",
+			wantIndexerResp: &response.IndexerGetNFTTokenDetailResponse{
+				Data: response.IndexerNFTTokenDetailData{
+					TokenID:           "1",
+					CollectionAddress: "0xb54FF1EBc9950fce19Ee9E055A382B1219f862f0",
+					Name:              "Portalhead #1",
+					Description:       "A Modern People Project.",
+					Amount:            "",
```

Comment:
should remove if it is zero value (similar with others)

Reply:
i can remove empty test inputs, but i think we should keep api response exact format even if they are empty